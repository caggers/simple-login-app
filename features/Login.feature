Feature: User Login
	I want to log in with my credentials

Scenario: See the login Page
	Given I am a User
	When the login page loads 
	Then I should see username and password inputs

Scenario: Enter User credentials
	Given I am any User
	When I fill in my username and password credentials
	When my credentials are submitted
	Then I see an error message

Scenario: Check User credentials for a registered User
	Given I am a registered User logging in
	When my credentials are sent for authentication
	When I click the submit button
	Then I can see the App




