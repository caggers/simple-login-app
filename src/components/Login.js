import React, { Component } from 'react';


class Login extends Component {
  constructor() {
    super();
    this.state = {
      username: '',
      pword: ''
    }
  }

  handleChange = (e) => {
    const val = e.target.value;
    const name = e.target.name;
    this.setState({
      [name]: val
    });
  }

  render() {
    return(
      <form>
        <input 
          className="input-username"
          name="username"
          onChange={this.handleChange} 
        />
        <input 
          className="input-password"
          name="pword"
          onChange={this.handleChange}  />
      </form>
    );
  }
}

export default Login;