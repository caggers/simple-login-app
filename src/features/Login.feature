Feature: User Login Page
	In order to access my homepage
	As a User
	I want to log in with my credentials
	And access the App

Scenario: See the Login Page
	Given I am a User
	When the login page loads 
	Then I should see username and password inputs

Scenario: Enter User credentials
	Given I am a registered User
	When I fill in my username and password credentials
	And I click submit
	Then my credentials should be submitted

Scenario: Check User credentials for a registered User
	Given I am a registered User logging in
	When my credentials are sent for authentication
	And they pass
	Then I am logged into the system
	And I can see the App
Examples: Valid
| username 	| password 	|
| admin 		| admin			|
| user			| user 			|

Scenario: Check User credentials for an unregistered User
	Given I am not registered user logging in
	When my invalid credentials are sent for authentication
	And they do not pass
	Then I see an error message
Examples: Invalid
| username 	| password 	|
| 1234			| 5678			|
| betty			| spagetti	|


