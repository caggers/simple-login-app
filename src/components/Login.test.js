import React from 'react';
import { shallow, mount } from 'enzyme';
import Login from './Login';

describe('Login', () => {
  const login = shallow(<Login />);

  it('renders correctly', () => {
    expect(login).toMatchSnapshot();
  });

  it('initialises a username and password in `state`', () => {
    expect(login.state()).toEqual({ username: '', pword: '' });
  });

  describe('User input', () => {
    const username = 'user';
    const pword = 'p@$$word';


    beforeEach(() => {
      login
        .find('.input-username')
        .simulate('change', { target: { value: username, name: 'username' } });
      login
        .find('.input-password')
        .simulate('change', { target: { value: pword, name: 'pword' } });
    });

    it('renders the inputs', () => {
      expect(login.find('.input-username')).toExist();
      expect(login.find('.input-password')).toExist();
    });

    it('when typing into the inputs it updates relevant fields in `state`', () => {
      expect(login.state().username).toEqual(username);
      expect(login.state().pword).toEqual(pword);
    });
  });

  // describe('Password Input', () => {
   
  //   beforeEach(() => {
  //     login
  //       .find('.input-password')
  //       .simulate('change', { target: { value: pword, name: 'pword' } });
  //   });

  //   it('renders the password input', () => {
      
  //   });

  //   it('when typing into the input it updates the password in `state`', () => {
      
  //   });
  // });
});
