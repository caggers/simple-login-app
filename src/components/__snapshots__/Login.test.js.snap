// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders correctly 1`] = `
<Login
  authenticateUser={[MockFunction]}
>
  <div
    className="login"
  >
    <h1>
      Login
    </h1>
    <form>
      <div
        className="form-row"
      >
        <label>
          Username:
          <input
            className="input-username"
            name="username"
            onChange={[Function]}
          />
        </label>
      </div>
      <div
        className="form-row"
      >
        <label>
          Password:
          <input
            className="input-password"
            name="pword"
            onChange={[Function]}
          />
        </label>
      </div>
      <div
        className="form-row"
      >
        <Button
          className="btn-submit"
          onClick={[Function]}
          text="Submit"
          type="submit"
        >
          <button
            className="btn-instance btn-submit"
            onClick={[Function]}
            type="submit"
          >
            Submit
          </button>
        </Button>
      </div>
    </form>
  </div>
</Login>
`;
